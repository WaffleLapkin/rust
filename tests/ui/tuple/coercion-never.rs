// This test checks if tuple elements are a coercion site or not.
// Note that the code here is a degenerate case, but you can get similar effects in real code, when
// unifying match arms, for example.
//
// See also coercion-slice.rs

fn main() {
    let _: ((),) = (loop {},);

    ((),) = (loop {},); //~ error: mismatched types

    let x = (loop {},);
    let _: ((),) = x; //~ error: mismatched types
}
