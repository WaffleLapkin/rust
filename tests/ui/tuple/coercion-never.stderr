error[E0308]: mismatched types
  --> $DIR/coercion-never.rs:10:6
   |
LL |     ((),) = (loop {},);
   |      ^^     ---------- this expression has type `(!,)`
   |      |
   |      expected `!`, found `()`
   |
   = note:   expected type `!`
           found unit type `()`

error[E0308]: mismatched types
  --> $DIR/coercion-never.rs:13:20
   |
LL |     let _: ((),) = x;
   |            -----   ^ expected `((),)`, found `(!,)`
   |            |
   |            expected due to this
   |
   = note: expected tuple `((),)`
              found tuple `(!,)`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
